### PR TITLE
[inductor] Misc compile time improvements

### PR DESCRIPTION
**Pull request by jansel**
This is a smaller improvement in compile time.  Roughly 5%.

### CODE REVIEW

File: torchinductor/config.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 benchmark_harness = True
 
 # control store vs recompute heuristic
-realize_reads_threshold = 4
+realize_reads_threshold = 5
```
Comment (jansel) [inline torchinductor/config.py:28]: I measured this being slightly faster on torchbench.

File: torchinductor/scheduler.py [hunk 2]
```diff
@@ -701,7 +701,7 @@ def topological_sort_schedule(self):
         def visit(n):
             if n not in seen:
                 seen.add(n)
-                for dep in n.unmet_dependencies:
+                for dep in sorted(n.unmet_dependencies, key=lambda d: d.name):
```
Comment (jansel) [inline torchinductor/scheduler.py:704]: This makes our output wrapper code deterministic again.

File: torchinductor/config.py (hunk 1 above)
Comment (jansel) [reply torchinductor/config.py]: I reverted this because it caused an illegal memory access on one timm model.

File: torchinductor/config.py [hunk 3]
```diff
@@ -73,9 +73,6 @@ class triton:
     # Use cudagraphs on output code
     cudagraphs = True
 
-    # Monkey patching to lower overheads
-    hackery = False
```
Comment (Chillee) [inline torchinductor/config.py:77]: Nice :)

File: torchinductor/ir.py [hunk 4]
```diff
@@ -3180,22 +3182,25 @@ def realize_hint(self):
         """
         Called on buffers we expect to be forced to realize later.
         """
-        if self.num_reads() > 1:
+        if isinstance(self.data, (Pointwise, Reduction)) and self.num_reads() > 1:
```
Comment (Chillee) [inline torchinductor/ir.py:3185]: When does this ever matter?

**Review by Chillee — APPROVED** [review]

File: torchinductor/ir.py (hunk 4 above)
Comment (jansel) [reply torchinductor/ir.py:3185]: It just avoids a call to `num_reads`, which is expensive.

**Pull request merged**